### DISCUSSION TITLE
404 after upgrading Seq

### DISCUSSION THREAD

**Discussion by surlean-dev** [datalust/seq-tickets#2264] [Help]
Hello,
I just upgrade my Seq installation to v2024.3.12250 and now it no longer works. The Seq service will not start. The log shows the following:

{"@t":"2024-09-05T15:38:36.8208051Z","@mt":"Seq {SeqVersion} running on OS {OSCaption}","SeqVersion":"2024.3.12250","OSCaption":"Microsoft Windows Server 2016 Datacenter","SourceContext":"Seq.Server.Features.Diagnostics.EnvironmentInformationWriter"}
{"@t":"2024-09-05T15:38:36.8811284Z","@mt":"Running in Windows Service host"}
{"@t":"2024-09-05T15:38:37.6679935Z","@mt":"Opening event store at {StreamPath}","StreamPath":"C:\\ProgramData\\Seq\\Stream\\stream.flare","SourceContext":"Seq.Engine.Cluster.StorageNodeCoordinator"}
{"@t":"2024-09-05T15:38:37.7704533Z","@mt":"Ingestion enabled","SourceContext":"Seq.Engine.Cluster.StorageNodeCoordinator"}
{"@t":"2024-09-05T15:38:37.7987702Z","@mt":"Opening metastore {MetastorePath}","MetastorePath":"C:\\ProgramData\\Seq\\Documents\\metastore.flare","SourceContext":"Quince.Storage.Native.NativeDocumentStore"}
{"@t":"2024-09-05T15:38:37.9076853Z","@mt":"Storage subsystem available"}
{"@t":"2024-09-05T15:38:38.0529968Z","@mt":"Error running the server application","@l":"Fatal","@x":"System.Net.Sockets.SocketException (10013): An attempt was made to access a socket in a way forbidden by its access permissions.\r\n   at System.Net.Sockets.Socket.UpdateStatusAfterSocketErrorAndThrowException(SocketError error, Boolean disconnectOnFailure, String callerName)\r\n   at System.Net.Sockets.Socket.DoBind(EndPoint endPointSnapshot, SocketAddress socketAddress)\r\n   at System.Net.Sockets.Socket.Bind(EndPoint localEP)\r\n   at Microsoft.AspNetCore.Server.Kestrel.Transport.Sockets.SocketTransportOptions.CreateDefaultBoundListenSocket(EndPoint endpoint)\r\n   at Microsoft.AspNetCore.Server.Kestrel.Transport.Sockets.SocketConnectionListener.Bind()\r\n   at Microsoft.AspNetCore.Server.Kestrel.Transport.Sockets.SocketTransportFactory.BindAsync(EndPoint endpoint, CancellationToken cancellationToken)\r\n   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Infrastructure.TransportManager.BindAsync(EndPoint endPoint, ConnectionDelegate connectionDelegate, EndpointConfig endpointConfig, CancellationToken cancellationToken)\r\n   at Microsoft.AspNetCore.Server.Kestrel.Core.KestrelServerImpl.<>c__DisplayClass28_0`1.<<StartAsync>g__OnBind|0>d.MoveNext()\r\n--- End of stack trace from previous location ---\r\n   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.AddressBinder.BindEndpointAsync(ListenOptions endpoint, AddressBindContext context, CancellationToken cancellationToken)\r\n   at Microsoft.AspNetCore.Server.Kestrel.Core.ListenOptions.BindAsync(AddressBindContext context, CancellationToken cancellationToken)\r\n   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.AddressBinder.EndpointsStrategy.BindAsync(AddressBindContext context, CancellationToken cancellationToken)\r\n   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.AddressBinder.BindAsync(ListenOptions[] listenOptions, AddressBindContext context, Func`2 useHttps, CancellationToken cancellationToken)\r\n   at Microsoft.AspNetCore.Server.Kestrel.Core.KestrelServerImpl.BindAsync(CancellationToken cancellationToken)\r\n   at Microsoft.AspNetCore.Server.Kestrel.Core.KestrelServerImpl.StartAsync[TContext](IHttpApplication`1 application, CancellationToken cancellationToken)\r\n   at Microsoft.AspNetCore.Hosting.WebHost.StartAsync(CancellationToken cancellationToken)\r\n   at Microsoft.AspNetCore.Hosting.WebHost.Start()\r\n   at Seq.ServiceProcess.ServerService.Start()"}
{"@t":"2024-09-05T15:38:38.0722481Z","@mt":"The Seq server could not be started","@l":"Fatal","@x":"System.Net.Sockets.SocketException (10013): An attempt was made to access a socket in a way forbidden by its access permissions.\r\n   at System.Net.Sockets.Socket.UpdateStatusAfterSocketErrorAndThrowException(SocketError error, Boolean disconnectOnFailure, String callerName)\r\n   at System.Net.Sockets.Socket.DoBind(EndPoint endPointSnapshot, SocketAddress socketAddress)\r\n   at System.Net.Sockets.Socket.Bind(EndPoint localEP)\r\n   at Microsoft.AspNetCore.Server.Kestrel.Transport.Sockets.SocketTransportOptions.CreateDefaultBoundListenSocket(EndPoint endpoint)\r\n   at Microsoft.AspNetCore.Server.Kestrel.Transport.Sockets.SocketConnectionListener.Bind()\r\n   at Microsoft.AspNetCore.Server.Kestrel.Transport.Sockets.SocketTransportFactory.BindAsync(EndPoint endpoint, CancellationToken cancellationToken)\r\n   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Infrastructure.TransportManager.BindAsync(EndPoint endPoint, ConnectionDelegate connectionDelegate, EndpointConfig endpointConfig, CancellationToken cancellationToken)\r\n   at Microsoft.AspNetCore.Server.Kestrel.Core.KestrelServerImpl.<>c__DisplayClass28_0`1.<<StartAsync>g__OnBind|0>d.MoveNext()\r\n--- End of stack trace from previous location ---\r\n   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.AddressBinder.BindEndpointAsync(ListenOptions endpoint, AddressBindContext context, CancellationToken cancellationToken)\r\n   at Microsoft.AspNetCore.Server.Kestrel.Core.ListenOptions.BindAsync(AddressBindContext context, CancellationToken cancellationToken)\r\n   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.AddressBinder.EndpointsStrategy.BindAsync(AddressBindContext context, CancellationToken cancellationToken)\r\n   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.AddressBinder.BindAsync(ListenOptions[] listenOptions, AddressBindContext context, Func`2 useHttps, CancellationToken cancellationToken)\r\n   at Microsoft.AspNetCore.Server.Kestrel.Core.KestrelServerImpl.BindAsync(CancellationToken cancellationToken)\r\n   at Microsoft.AspNetCore.Server.Kestrel.Core.KestrelServerImpl.StartAsync[TContext](IHttpApplication`1 application, CancellationToken cancellationToken)\r\n   at Microsoft.AspNetCore.Hosting.WebHost.StartAsync(CancellationToken cancellationToken)\r\n   at Microsoft.AspNetCore.Hosting.WebHost.Start()\r\n   at Seq.ServiceProcess.ServerService.Start()\r\n   at Seq.ServiceProcess.SeqWindowsService.<>c__DisplayClass2_0.<.ctor>b__0()"}

Thanks,
Randy Blackmond

**Comment by nblumhardt** [comment]
Hi Randy, thanks for your message.

At the Windows Start menu, you can open Seq Service Administration and step through configuration again; there's a known issue in earlier 2024.x installers that caused some later upgrades to fail, and reconfiguring the service should sort this out.

If not, just let me know. Hope this helps!

Nick

**Reply by ranbla** [reply]
Hi Nick,

I tried that but it didn't resolve the issue. After running the Seq Service Administration and clicking on the Browse Seq button, I get this:

![Seq_404_After_Upgrade](https://github.com/user-attachments/assets/269953c7-bc2e-4417-b512-7281663a09be)

Randy

**Reply by nblumhardt** [reply]
Ah, I see! Thanks for the follow-up. Using Seq at a nested path (`/seq`) requires choosing the HTTP.sys web server in the Seq Service Administration config dialog. There should be a link "Configure HTTP.sys instead" on the page that accepts URLs and paths.

A bug in an earlier 2024.x build causes some service settings to be lost on upgrade; your system would earlier have been using HTTP.sys, but the new default is Kestrel. The bug is now fixed, but the fix can't be applied retroactively, unfortunately.

It's a good idea to also double-check that the "storage path" setting points to the correct folder containing your original Seq data. Hope this helps!

**Reply by ranbla** [reply]
That was it! It's running normally now. Thanks a lot!

Randy

**Reply by nblumhardt** [reply]
Awesome 👍 - you're welcome!
